.github/workflows/ci_visualization.yml

```diff
@@ -0,0 +1,45 @@
+# This is a basic workflow to help you get started with Actions
+
+name: ci_visualization
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: "*"
+  pull_request:
+    branches: [main]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  SRC_DIR: "src/implot_visualization"
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Init submodules
+        run: git submodule update --init --recursive
+
+      - name: Install emscripten
+        uses: mymindstorm/setup-emsdk@v11
+        with:
+          version: 2.0.14
+          # actions-cache-folder: "emsdk-cache"
+
+      - name: Verify emscripten
+        run: emcc -v
+
+      - name: Build
+        shell: bash
+        run: emcmake cmake -S ${{env.SRC_DIR}} -B ${{env.SRC_DIR}}/build && (cd ${{env.SRC_DIR}}/build && emmake make -j 20)
```

Comment:
The build dir does not need to be inside the source directory, so ./build should be sufficient.

-j20 will not do any good, github runners only have 2 cores.

Splitting the cmake generate and build step into 2 separate jobs will make the log a bit easier to navigate and generate better CI faliures that directly tell what went wrong.

Reply:
I tried to do it with ./build but this resulted in an error "No CMakeLists found in source directory". With ./build the runner searches for the CMakeLists directly in the pulsed-power-ml folder.

Reply:
Hm that's weird, the argument to -B should not influence where the CMakeLists are searched. But nothing to stall this for.